solver/src/oneinch_solver.rs

```diff
@@ -158,51 +156,15 @@ impl Interaction for Swap {
     }
 }
 
-/// Maximum number of sell orders to consider for settlements.
-///
-/// This is mostly out of concern to avoid rate limiting and because 1Inch
-/// requests take a non-trivial amount of time.
-const MAX_SETTLEMENTS: usize = 5;
-
 #[async_trait::async_trait]
-impl Solver for OneInchSolver {
-    async fn solve(&self, liquidity: Vec<Liquidity>, _gas_price: f64) -> Result<Vec<Settlement>> {
-        let mut sell_orders = liquidity
-            .into_iter()
-            .filter_map(|liquidity| match liquidity {
-                Liquidity::Limit(order) if order.kind == OrderKind::Sell => Some(order),
-                _ => None,
-            })
-            .collect::<Vec<_>>();
-
-        // Randomize which orders we take, this prevents this solver "getting
-        // stuck" on bad orders.
-        if sell_orders.len() > MAX_SETTLEMENTS {
-            sell_orders.shuffle(&mut rand::thread_rng());
-        }
-
+impl SingleOrderSolving for OneInchSolver {
+    async fn settle_order(&self, order: LimitOrder) -> Result<Settlement> {
+        ensure!(
+            order.kind == OrderKind::Sell,
+            "1Inch only supports sell orders"
+        );
```

Comment:
Any particular reason we are seeing single order solving pop up again here? These changes were already introduced elsewhere. I think you may want to base this on top of that.

Reply:
Had to rebase after merging the base branch. Should be fixed now